pyproject.toml

```diff
@@ -6,3 +6,163 @@ requires = [
     "pybind11>=2.6",
     "setuptools_scm>=7",
 ]
+[tool.ruff]
+exclude = [
+    ".git",
+    "build",
+    "doc/gallery",
+    "doc/tutorials",
+    "tools/gh_api.py",
+    ".tox",
+    ".eggs",
+]
+ignore = [
+    "D100",
+    "D101",
+    "D102",
+    "D103",
+    "D104",
+    "D105",
+    "D106",
```

Comment:
You are right, but that is how they are in the flake8 config, sooo???

My best guess is that that was originally put that way because those are rules we would _like_ to enforce, but have too many failures to individually enumerate them yet.

I ran `flake8-to-ruff` (a tool distributed separately on pypi by the ruff devs)
The only rule selection difference I made explicitly was to ignore D90 (mccabe cyclomatic complexity) because that didn't actually do anything on flake8 (the config value was null).

Outside of that the handful of rules that are not currently supported by ruff are added to "external" (There may be cause to add to ignores later as well, didn't look too closely at those because they can't be configured as ignores yet anyway)

It did not respect order of the per file ignores so I re-sorted them because that made it much easier to navigate.

Reply:
Okay, I tried it out, and selecting "D" was largely simply fine...

Looking back at the flake8 config, it contains some comments which indicate that the select was all rules for the numpy convention. But then ignored are all the rules we break regularly.

I already did configure it to use numpy style docstrings, which automatically disables several of the ignored rules (without having to specify) Thus selecting all "D" rules and only ignoring has the same effect, and cleans up the config quite a bit.

Additionally, the following rules from the ignore list had only a single occurrence, so I may as well fix that occurrence and enforce the rule:

- D207: Docstring is under-indented 
   - just a single missing space in the closing `"""` for one docstring
- D403: First word of the first line should be properly capitalized
   - No occurrences, if no objection to enforcing this going forward, may as well enable
- E722: do not use bare except, specify exception instead
   - single occurence in tests, added a noqa comment (could alternatively add to per file ignore if we wish)


Additionally, selecting all pydocstyle (`D`) enabled the previously unselected D419: Docstring is empty.

There was a single occurrence of an empty docstring, in scale.py:LinearScale, which has a comment stating that the method is there explicitly to prevent inherited docstring.
 - noqa comment added, enforcing the rule otherwise
 
The N error codes (enforcing naming conventions) that were included in the ignore list were also never selected (and selecting "N" was rather noisy even with the ignores).
These have been removed

Reply:
I think incrementally adding linter checks when appropriate is probably going to be easier, so this sounds fine to me.